docs/src/using-turing/guide.md

```diff
@@ -545,6 +545,41 @@ The `Gibbs` sampler can be used to specify unique automatic differentation backe
 
 For more details of compositional sampling in Turing.jl, please check the corresponding [paper](http://proceedings.mlr.press/v84/ge18b.html).
 
+### Working with filldist and arraydist
+
+Turing provides `filldist(dist::Distribution, n::Int)` and `arraydist(dists::AbstractVector{<:Distribution})` as a simplified interface to construct product distributions, e.g., to model a set of variables that share the same structure but vary by group.
```

Comment:
Just a heads up, rather sooner than later they will be integrated and generalized in Distributions: https://github.com/JuliaStats/Distributions.jl/pull/1391 I am working on a PR to DistributionsAD that would fix the errors mentioned in the last comments; then we can switch `product_distribution` completely from `Product` to `ProductDistribution`.

Reply:
Lovely improvements - I look forward to the new Distributions release!